src/main/java/io/github/cottonmc/libcd/mixin/MixinAdvancementRewardsBuilder.java

```diff
@@ -0,0 +1,37 @@
+package io.github.cottonmc.libcd.mixin;
+
+import com.google.common.collect.Maps;
+import io.github.cottonmc.libcd.impl.CustomRewardsBuilder;
+import io.github.cottonmc.libcd.impl.CustomRewardsSettingsUtils;
+import net.minecraft.advancement.AdvancementRewards;
+import net.minecraft.util.Identifier;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.Map;
+
+@Mixin(AdvancementRewards.Builder.class)
+public class MixinAdvancementRewardsBuilder implements CustomRewardsBuilder {
+    private final Map<Identifier, Object> customRewardsSettings = Maps.newHashMap();
+
+    @Inject(method = "build", at = @At("TAIL"), cancellable = true)
+    public void onBuild(CallbackInfoReturnable<AdvancementRewards> cir) {
+        CustomRewardsSettingsUtils value = (CustomRewardsSettingsUtils) cir.getReturnValue();
+        value.addAllRewardsSettings(customRewardsSettings);
+        cir.setReturnValue((AdvancementRewards) value);
```

Comment:
I don't think this should be necessary, since you're just mutating the return value and not replacing it.